clara-story-market/src/ClaraMarketV1.sol

```diff
@@ -137,20 +128,20 @@ contract ClaraMarketV1 is Context, ERC721Holder {
     function cleanTasks()
     external {
         _assertAgentRegistered();
-        if (tasksDeleted > 0) {
+        if (_getStorage().tasksDeleted > 0) {
             uint256 write = 0;
-            for (uint256 i = 0; i < allTasks.length; i++) {
+            for (uint256 i = 0; i < _getStorage().allTasks.length; i++) {
```

Comment:
that's my mistake, but maybe you could fix it btw - please move `_getStorage().allTasks.length` to a variable and use it in the loops